src/coreclr/debug/daccess/request_common.h

```diff
@@ -29,35 +29,27 @@ DPTR(T) Dereference(DPTR(T*) ptr)
 // Indexes into the global heap table, returning a TADDR to the requested
 // heap instance.
 inline TADDR
-HeapTableIndex(DPTR(opaque_gc_heap**) heaps, size_t index)
+HeapTableIndex(DPTR(unused_gc_heap**) heaps, size_t index)
 {
-    DPTR(opaque_gc_heap*) heap_table = Dereference(heaps);
-    DPTR(opaque_gc_heap*) ptr = TableIndex(heap_table, index, sizeof(void*));
+    DPTR(unused_gc_heap*) heap_table = Dereference(heaps);
+    DPTR(unused_gc_heap*) ptr = TableIndex(heap_table, index, sizeof(void*));
     return Dereference(ptr).GetAddr();
 }
 
-// Indexes into a given generation table, returning a DPTR to the
-// requested element (the element at the given index) of the table.
-inline DPTR(dac_generation)
```

Comment:
This is meant to reduce complexity at a small cost of performance. If I were to do that lazily, I have to either introduce a class that performs the loading lazily on access (one could imagine how complex it would be to make the macros to generate such a class) or to requires all the downstream usages to use the `LOAD` macro (one could imagine how messy that would look). 

The cost of not lazily loading them was just reading a few more bytes here, which is very probably cached anyway. IMO, this is a very reasonable trade-off to make.

Reply:
Just wanted to make sure dragons did not live here. Makes sense given the offset approach.